pkg/apis/allocation/v1/gameserverallocation.go

```diff
@@ -226,6 +323,68 @@ func (s *GameServerSelector) Validate(field string) ([]metav1.StatusCause, bool)
 		}
 	}
 
+	if !runtime.FeatureEnabled(runtime.FeatureCountsAndLists) && (s.Counters != nil || s.Lists != nil) {
```

Comment:
I'm fine with this but just want to call out this is a behavior change for customer. I.e. for the other existing selectors, if they are not featureEnabled, we just silently ignore them when performing Match without throwing any error back.

Reply:
I'm personally for more verbose help from the code, since it's easier to track down bugs than with silent failures. Do we have a pattern for logging warnings even if we don't error?

Reply:
Actually, I think the GSA is being inconsistent with Fleets and GameServers -- on those we do actively throw a validation error if you attempt to use a field that is behind a feature flag.

So I would argue, let's set a consistency here that we should validate feature flagged fields and continue doing it forward. 

Sound good?

Reply:
SGTM

Reply:
SGT,